### PR TITLE
gh-67230: document new csv quoting modes in whatsnew (gh-103491)

### PR DESCRIPTION
**Pull request by smontanaro**
Add blurb about recent change to csv.

* Issue: gh-67230

### CODE REVIEW

File: Doc/whatsnew/3.12.rst [hunk 1]
```diff
@@ -253,6 +253,13 @@ asyncio
 * :func:`asyncio.wait` now accepts generators yielding tasks.
   (Contributed by Kumar Aditya in :gh:`78530`.)
 
+csv
+---
+
+* Added :data:`QUOTE_NOTNULL` and :data:`QUOTE_STRINGS` flags to
```
Comment (merwok) [inline Doc/whatsnew/3.12.rst]:
```suggestion
* Add :data:`~csv.QUOTE_NOTNULL` and :data:`~csv.QUOTE_STRINGS` flags to
```

File: Doc/whatsnew/3.12.rst [hunk 2]
```diff
@@ -253,6 +253,13 @@ asyncio
...
+  provide finer control treatment of ``None`` and empty strings by
+  :class:`reader` and :class:`writer` objects.
```
Comment (merwok) [inline Doc/whatsnew/3.12.rst]:
```suggestion
  :class:`~csv.reader` and :class:`~csv.writer` objects.
```

**Review by merwok — CHANGES_REQUESTED** [review]

File: Doc/whatsnew/3.12.rst [hunk 3]
```diff
@@ -253,6 +253,13 @@ asyncio
 * :func:`asyncio.wait` now accepts generators yielding tasks.
   (Contributed by Kumar Aditya in :gh:`78530`.)
 
+csv
+---
+
+* Add :data:`~csv.QUOTE_NOTNULL` and :data:`~csv.QUOTE_STRINGS` flags to
+  provide finer control treatment of ``None`` and empty strings by
```
Comment (merwok) [inline Doc/whatsnew/3.12.rst]: control or treatment? 😉

Comment (smontanaro) [reply Doc/whatsnew/3.12.rst]: Thanks. "control", not "treatment."

I have never used this edit-thru-github feature before and can't figure out how to edit the file again. Also, I see the need for adding "csv." to the data references. What is the tilde for?

Comment (merwok) [reply Doc/whatsnew/3.12.rst]: tilde will make sphinx display only the last part of the dotted name

Comment (merwok) [reply Doc/whatsnew/3.12.rst]: quick edit on github PR:
- go to files tab
- hover over a line
- click `+` button next to line number to comment
- click `±` toolbar button to add a so-called suggestion in your comment
- edit the code in the suggestion, check the preview to see the diff
- publish comment
- commit the suggestion, or leave for the reviewer

**Comment by smontanaro** [comment]
This interface is horrible. I see a change was rewuested by @merwok but can't see what it is. So...

I have made the requested changes; please review again.

**Comment by merwok** [comment]
> I see a change was requested by merwok but can't see what it is. So...

Ah! «requested changes» is a review status (others are «approved» and «commented»)
Comments can be standalone or part of a review; I used the latter mode, made a few comments (some being change suggestions) and published my review with the «changes requested» status.  Then I applied some of my suggested changes directly.  So you saw «changes requested», some hidden comments («resolved» by the application of the suggestion), and one comment about the wording.

**Comment by samwyse** [comment]
@smontanaro looks good! I tried to do this myself last night, but I was at my kid's soccer practice and discovered that the official iPhone GitHub app does not have all the functionality of the web GUI. 😢

**Comment by chaeynz** [comment]
@smontanaro were you responsible for the csv.QUOTE_NULL?

You're an absolute legend!
I am working a project where I need exactly this, thank you so much! :)

**Comment by smontanaro** [comment]
Not exactly. Someone else suggested it years ago. I tend to catch most proposed changes to the csv module...

**Review by KNGSJ — COMMENTED** [review]
:)

**Pull request merged**